Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1310,23 +1304,34 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, unsigned int dstFo
     copy_to_vram = false;
   }
 
-  // Invalidate all textures that overlap the range of our efb copy.
-  // Unless our efb copy has a weird stride, then we mark them to check for partial texture updates.
-  // TODO: This also invalidates partial overlaps, which we currently don't have a better way
-  //       of dealing with.
-  bool invalidate_textures = dstStride == bytes_per_row || !copy_to_vram;
+  // Invalidate all textures, if they are either fully overwritten by our efb copy, or if they
+  // have a different stride than our efb copy. Textures with the same stride and that are partly
+  // overwritten are marked to check them for partial texture updates.
+  // TODO: The logic to detect a full overlap of strided efb copies is not 100% accurate
+  bool strided_efb_copy = dstStride != bytes_per_row;
   auto iter = FindOverlappingTextures(dstAddr, covered_range);
   while (iter.first != iter.second)
   {
     TCacheEntry* entry = iter.first->second;
     if (entry->OverlapsMemoryRange(dstAddr, covered_range))
     {
-      if (invalidate_textures)
+      u32 overlap_range = std::min(entry->addr + entry->size_in_bytes, dstAddr + covered_range) -
+                          std::max(entry->addr, dstAddr);
+      if (entry->memory_stride != dstStride || !copy_to_vram ||
+          (!strided_efb_copy && entry->size_in_bytes == overlap_range) ||
```

Comment:
I don't know what you mean. strided_efb_copy means that the currently generated efb copy has a weird stride, like the 32x32 efb copies in new super mario bros that contain the spinning coins, and that are used to update that 1024x1024 texture. Partial updates checks if the stride of the efb copy and the texture match, otherwise it refuses to update the texture. So entry->memory_stride != dstStride just checks if the efb copy and texture are compatible and if not, it's removed.